tests/test_interpolatedimage.py

```diff
@@ -1532,9 +1532,18 @@ def test_ii_shoot():
             rtol = 1.e-7
         assert np.isclose(added_flux, obj.flux, rtol=rtol)
         assert np.isclose(im.array.sum(), obj.flux, rtol=rtol)
-        photons2 = obj.makePhot(poisson_flux=False, rng=rng)
+        photons2 = obj.makePhot(poisson_flux=False, rng=rng.duplicate())
         assert photons2 == photons, "InterpolatedImage makePhot not equivalent to drawPhot"
 
+        # Can treat as a convolution of a delta function and put it in a photon_ops list.
+        delta = galsim.DeltaFunction(flux=flux)
+        psf = galsim.InterpolatedImage(image_in, x_interpolant=interp, scale=3.3, flux=1)
+        photons3 = delta.makePhot(poisson_flux=False, rng=rng.duplicate(), photon_ops=[psf],
+                                  n_photons=len(photons))
+        np.testing.assert_allclose(photons3.x, photons.x)
```

Comment:
Why is this one (and a few others) `allclose` but not `==` like the rest?

Reply:
I can't remember all the details now, but some of the calculations did things in slightly different orders, so things could be different at the 1.e-15 level.  Others involved different functions doing precisely the same calculations, so they were exactly equal.  I started with all of them as `photonsx == photons`, but this must have been one that was basically the same answer, but not precisely identical.  So I had to pull it apart and check the components separately.